src/wp-includes/script-loader.php

```diff
@@ -3468,3 +3474,124 @@ function wp_remove_surrounding_empty_script_tags( $contents ) {
 		);
 	}
 }
+
+/**
+ * Adds hooks to load block styles on demand in classic themes.
+ *
+ * @since 6.9.0
+ */
+function wp_load_classic_theme_block_styles_on_demand() {
+	if ( wp_is_block_theme() ) {
+		return;
+	}
+
+	/*
+	 * Make sure that wp_should_output_buffer_template_for_enhancement() returns true even if there aren't any
+	 * `wp_template_enhancement_output_buffer` filters added, but do so at priority zero so that applications which
+	 * wish to stream responses can more easily turn this off.
+	 */
+	add_filter( 'wp_should_output_buffer_template_for_enhancement', '__return_true', 0 );
+
+	if ( ! wp_should_output_buffer_template_for_enhancement() ) {
+		return;
+	}
+
+	/*
+	 * Load separate block styles so that the large block-library stylesheet is not enqueued unconditionally,
+	 * and so that block-specific styles will only be enqueued when they are used on the page.
+	 */
+	add_filter( 'should_load_separate_core_block_assets', '__return_true', 0 );
+
+	// Also ensure that block assets are loaded on demand (although the default value is from should_load_separate_core_block_assets).
+	add_filter( 'should_load_block_assets_on_demand', '__return_true', 0 );
+
+	// Add hooks which require the presence of the output buffer. Ideally the above two filters could be added here, but they run too early.
+	add_action( 'wp_template_enhancement_output_buffer_started', 'wp_hoist_late_printed_styles' );
+}
+
+/**
+ * Adds the hooks needed for CSS output to be delayed until after the content of the page has been established.
+ *
+ * @since 6.9.0
+ *
+ * @see wp_load_classic_theme_block_styles_on_demand()
+ * @see _wp_footer_scripts()
+ */
+function wp_hoist_late_printed_styles() {
+
+	// Skip the embed template on-demand styles aren't relevant, and there is no wp_head action.
+	if ( is_embed() ) {
+		return;
+	}
+
+	// Determine priority at which styles are printed at wp_head. Short-circuit if it was unhooked.
+	$wp_head_priority = has_action( 'wp_head', 'wp_print_styles' );
+	if ( false === $wp_head_priority ) {
+		return;
+	}
+
+	/*
+	 * While normally late styles are printed, there is a filter to disable prevent this, so this makes sure they are
+	 * printed. Note that this filter was intended to control whether to print the styles queued too late for the HTML
+	 * head. This filter was introduced in <https://core.trac.wordpress.org/ticket/9346>. However, with the template
+	 * enhancement output buffer, essentially no style can be enqueued too late, because an output buffer filter can
+	 * always hoist it to the HEAD.
+	 */
+	add_filter( 'print_late_styles', '__return_true', PHP_INT_MAX );
```

Comment:
Better to simply use a high number (I chose 100 pretty much randomly).

If a dev needs to follow the core function, they'll end up having to do `add_filter( 'print_late_styles', 'my_filter', PHP_INT_MAX + 1 );` and it becomes a race to the bottom...

```suggestion
	add_filter( 'print_late_styles', '__return_true', 100 );
```